js/packages/mobile-wallet-adapter-walletlib/android/src/main/java/com/solanamobile/mobilewalletadapterwalletlib/reactnative/SolanaMobileWalletAdapterWalletLibModule.kt

```diff
@@ -159,147 +142,195 @@ class SolanaMobileWalletAdapterWalletLibModule(val reactContext: ReactApplicatio
 
     /* Generic Request functions */
     @ReactMethod
-    fun cancelRequest() {
-        Log.d(TAG, "Cancelled request");
-        (request as? ScenarioRequest)?.let { scenarioRequest ->
+    fun cancelRequest(sessionId: String, requestId: String) {
+        Log.d(TAG, "Cancelled request $requestId");
+        (pendingRequests.remove(requestId) as? ScenarioRequest)?.let { scenarioRequest ->
             scenarioRequest.cancel();
         }
     }
 
     /* AuthorizeDapp Request */
     @ReactMethod
-    fun completeWithAuthorize(publicKey: ReadableArray, accountLabel: String?, walletUriBase: String?, authorizationScope: ReadableArray?) {
-        Log.d(TAG, "completeWithAuthorize: authorized public key = $publicKey")
-        (request as? MobileWalletAdapterRemoteRequest.AuthorizeDapp)?.request?.let { authRequest ->
-            authRequest.completeWithAuthorize(
-                convertFromReactByteArray(publicKey),
-                accountLabel,
-                null, // walletUriBase,
-                null, // convertFromReactByteArray(authorizationScope)
-            )
+    fun completeWithAuthorize(
+        sessionId: String, 
+        requestId: String, 
+        publicKey: ReadableArray, 
+        accountLabel: String?, 
+        walletUriBase: String?, 
+        authorizationScope: ReadableArray?
+    ) = launch {
+        checkSessionId(sessionId) {
+            Log.d(TAG, "completeWithAuthorize: authorized public key = $publicKey")
+            (pendingRequests.remove(requestId) as? MobileWalletAdapterRemoteRequest.AuthorizeDapp)?.request?.let { authRequest ->
```

Comment:
Question: What happens if the type check here fails for some reason? 

We've already executed `pendingRequests.remove(requestId). Does that get lost forever in the fail case? Is that even an issue?

Reply:
yeah so you are correct that the remove will still occur no matter what. Probably would be safer to only remove if we actually complete the request. I will update.